### PR TITLE
Vithu dao creation

### PR DESCRIPTION
**Pull request by vithushon**
## Motivation and Context

This branch creates a DataAccessObject specific to our Entities. It solves the problem of us not having a relevant DAO.

## Your Changes

Creation of mongo_data_access which includes three repositories implementing MongoRepository allowing access to the three distinct collections in pdflex-db. Creation of MongoDataAccessObject that will implement our usecase DataAccessInterfaces (and consequently implementing the methods in each of our DAIs) to provide functionality in accessing the database. Creating of ViewAllClaims example use case with proper Clean Architecture structure to serve as an example for creating use cases with springboot/mongo. Some slight changes to entities to better fit with the data in our database - I suspect these will need to be edited a bit more as we delve further into our entities and use cases. One major change is that the FlexApplication class was moved outside of the app directory. This allowed for the code to run correctly.

**Type of change** (select all that apply):

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (internal change to codebase, without changing functionality)

## Notes for Reviewer

Currently the only use case implemented with the pdflex-db is the view all claims use case. You can see this works the same by ensuring the database in your env file is set to pdflex-db and then running the FlexApplication.java class. This should allow you to access localhost:8080/claims which will display a list of the claims in our 'claims' collection in a JSON format.

## Checklist

- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: flex/src/main/java/consonants/flex/use_case/view_all_claims/ViewAllClaimsInteractor.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+
+@RestController
+public class ViewAllClaimsInteractor implements ViewAllClaimsInputBoundary{
+
+    @Autowired
+    final ViewAllClaimsDataAccessInterface claimsDataAccessObject;
```
Comment (bensandoval) [inline flex/src/main/java/consonants/flex/use_case/view_all_claims/ViewAllClaimsInteractor.java:15]: final?

File: flex/src/main/java/consonants/flex/use_case/view_all_claims/ViewAllClaimsInteractor.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    @Autowired
+    final ViewAllClaimsOutputBoundary outputBoundary;
```
Comment (bensandoval) [inline flex/src/main/java/consonants/flex/use_case/view_all_claims/ViewAllClaimsInteractor.java:18]: final?

**Review by bensandoval — COMMENTED** [review]
See comments

File: flex/src/main/java/consonants/flex/use_case/view_all_claims/ViewAllClaimsInteractor.java (hunk 2 above)
Comment (bensandoval) [reply flex/src/main/java/consonants/flex/use_case/view_all_claims/ViewAllClaimsInteractor.java:18]: also outputBoundary should be renamed to viewAllClaimsPresenter

Comment (bensandoval) [reply flex/src/main/java/consonants/flex/use_case/view_all_claims/ViewAllClaimsInteractor.java:18]: otherwise it all looks good!

**Comment by lana-w** [comment]
Review Process:

Confirmed View All Claims Use Case is successfully implemented; did so by connecting a front-end page under the create-view-all-claims branch to test. 
DAO: connects successfully to database and methods return values as required. Note for future improvements, clarity in method names can be improved (findClient -> findClientById), but this will be revisited in later stages when consolidating use cases.

**Pull request merged**